src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataViewModel.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    using SymbolKind = LanguageServer.Protocol.SymbolKind;
+
+    /// <summary>
+    /// A ViewModel over <see cref="DocumentSymbolData"/>
+    /// The only items that are mutable on this type are <see cref="IsExpanded"/> and <see cref="IsSelected"/>.
+    /// It is expected that these can be modified from any thread with INotifyPropertyChanged notifications
+    /// being marshalled to the correct thread by WPF if there needs to be a change to the visual presentation.
+    /// </summary>
+    internal sealed class DocumentSymbolDataViewModel : INotifyPropertyChanged, IEquatable<DocumentSymbolDataViewModel>
+    {
+        private readonly DocumentSymbolData _data;
+
+        public string Name => _data.Name;
```

Comment:
nit: just expose the data.  callers can just see through that intead of having to lift all the same properties here.

Reply:
comment still applies.